### DISCUSSION TITLE
MET configuration with grib2

### DISCUSSION THREAD

**Discussion by scatt11** [dtcenter/METplus#1269] [Configuration] [answered]
Hello MET team,

I am having issues using Grib2 files with MET V10.0.0 [#1245](https://github.com/dtcenter/METplus/discussions/1245).  The issue is believed to be linked to installation/configuration.  I have read up on installing with grib2 functionality, and it seems I have done so.  I am using the compile_MET_all.sh script and a configuration file which can also be referenced in [#1245](https://github.com/dtcenter/METplus/discussions/1245).  I refrain from explicitly setting MET_GRIB2CLIB in order for the script to install and link GRIB2C.  My log messages for successful installation are attached.  
[log.log](https://github.com/dtcenter/METplus/files/7542346/log.log)

What needs to be changed to ensure that I configure with grib2 compatibility?

**Comment by jprestop** [comment, marked as answer]
HI @scatt11.

From your log file it looks like everything is set correctly to have GRIB enabled..  I see that --enable-grib2 is being set at the configure step.  Have you tried running with the the new MET binaries that were created from this build?  If you have and there are still problems, please send the make.log file from the G2CLIB build:

```
Compiling G2CLIB at Mon Nov 15 22:16:47 CET 2021
cd /home/computeadmin/meteo/Libraries/MET_install/external_libs/g2clib/g2clib-1.6.0
make > make.log 2>&1
```

along with the config.log, make.log, and make_test.log files from the MET build (in /home/computeadmin/meteo/Libraries/MET_install/met-10.0.0).  Please also send the command line you are running along with the output from the run.  Thanks!

**Reply by scatt11** [reply]
Hello @jprestop, yes I have attempted to run with the new build and still receiving the same error observed in [#1245](https://github.com/dtcenter/METplus/discussions/1245).  I have supplied the files as you requested.  

Most recent configuration command and I have included log.log:
./compile_MET_all.sh install_met_env.ripper 2>&1 | tee log.log

[make.log](https://github.com/dtcenter/METplus/files/7548510/make.log)
[make_g2clib.log](https://github.com/dtcenter/METplus/files/7548511/make_g2clib.log)
[make_test.log](https://github.com/dtcenter/METplus/files/7548512/make_test.log)
[config.log](https://github.com/dtcenter/METplus/files/7548513/config.log)
[log.log](https://github.com/dtcenter/METplus/files/7548515/log.log)

Thanks in advance.

**Reply by jprestop** [reply]
Thanks @scatt11. I took a quick look at the files and everything seems to be in order.  There were no errors in the make_test.log file so all of the test for the GRIB data seem to have run fine.  

Can you please give the following command a try?
```
plot_data_plane st4_conus.2021060700.01h.grb2 st4_conus.2021060700.01h.ps 'name="APCP"; level="A01";' -title "GRIB2 Stage IV 1-hr precip" -v 7
```

Please send all of the output from the screen when you run that command.  Thank you!

**Reply by scatt11** [reply]
I have copied my data file to this directory for simplicity and run the command.  Output is shown below.  I don't think I mentioned that you assisted with a configuration issue previously.  I doubt it will be responsible for the error, but I will reference the ticket anyway [#1157](https://github.com/dtcenter/METplus/discussions/1157)

~/meteo/Libraries/MET_install/met-10.0.0/src/tools/other/plot_data_plane$ ./plot_data_plane st4_conus.2021060700.01h.grb2 st4_conus.2021060700.01h.ps 'name="APCP"; level="A01";' -title "GRIB2 Stage IV 1-hr precip" -v 7
DEBUG 1: Opening data file: st4_conus.2021060700.01h.grb2
DEBUG 4: Met2dDataFileFactory::new_met_2d_data_file() -> created new Met2dDataFile object of type "FileType_Gb2".
*** stack smashing detected ***: <unknown> terminated
Aborted (core dumped)

**Reply by jprestop** [reply]
Hi @scatt11. 

> I don't think I mentioned that you assisted with a configuration issue previously. I doubt it will be responsible for the error, but I will reference the ticket anyway #1157

Thanks for mentioning that.  I do believe the issues are unrelated, as you suspected.

I'm a bit stumped by this one.  It seems you have installed the GRIB functionality successfully, and I cannot reproduce this error, so it is difficult to determine what is going on.  I need to consult with some other members on my team (@JohnHalleyGotway) to try to figure out where to go from here.  I'll follow up once I have more information.

**Reply by jprestop** [reply]
Hi @scatt11.  Could you please send your install_met_env.ripper file?  Thanks!

**Reply by scatt11** [reply]
Here you have it. Thanks
[install_met_env.ripper.txt](https://github.com/dtcenter/METplus/files/7566677/install_met_env.ripper.txt)

**Reply by jprestop** [reply]
@scatt11 Thank you!

I see that you have the following set:
```
export SET_D64BIT=TRUE
export CFLAGS="-D_64BIT_"
```

Could you please remove both of those settings from the file?  We want to try compiling without the D_64BIT flag set.  Then, please rerun the compile script so that the external libraries and MET will be recompiled.  Please let us know how it goes.

**Reply by scatt11** [reply]
@jprestop The compile was successful and plot_data_plane now completes successfully.  Thanks for all the help and please let me know if you need further information for tracking the source of this issue.

**Reply by jprestop** [reply]
That's great news!  Thank you for letting us know and for marking this Discussion as answered.
